doc/tutorials/basic/README.md

```diff
@@ -0,0 +1,303 @@
+    '--data_dir=/tmp/mnist',
+    '--num_gpus=1',
+    '--train_epochs=1',
+    '--model_dir=$(MODEL_DIR)',
+  ],
+  [...]
+```
+
+Note how we changed the `--num_gpu` flag to 1 and replaced the `accelerator` field with  `gpus.teslaV100`. Run `jsonnet` to inspect the update Kubernetes `Job`:
+
+```bash
+$ jsonnet --jpath . --string --ext-str gcs-bucket=$GCS_BUCKET doc/tutorials/basic/mnist-gpu.jsonnet
+```
+
+Notice the following lines in the output:
+
+```
+   [...]
+      "resources":
+         "limits":
+         "nvidia.com/gpu": 1
+   [...]
+   "nodeSelector":
+     "cloud.google.com/gke-accelerator": "nvidia-tesla-v100"
+```
+
+This will run the test on a node with at least 1 Nvidia Tesla V100 GPU available. Run the `Job` with `jsonnet` and `kubectl`
+
+```bash
+$ jsonnet --jpath . --string --ext-str gcs-bucket=$GCS_BUCKET doc/tutorials/basic/mnist-gpu.jsonnet | kubectl create -f -
+```
+
+Use `kubectl` or Cloud Console to check the job results.
+
+## Updating the test to use TPUs (optional)
+
+Similarly, we can configure the test to use TPUs too. See the following changes in `mnist-tpu.jsonnet`:
+
+```jsonnet
+local base = import 'templates/base.libsonnet';
+local tpus = import 'templates/tpus.libsonnet';
+
+local mnist = base.BaseTest {
+  [...]
+  tpuSettings+: {
+    softwareVersion: '2.9.1',
+  },
+  accelerator: tpus.v2_8,
+  [...]
+  command: [
+    'python3',
+    '-m',
+    'official.legacy.image_classification.mnist_main',
+    '--distribution_strategy=tpu',
+    '--data_dir=$(MNIST_DIR)',
+    '--train_epochs=1',
+    '--model_dir=$(MODEL_DIR)',
+  ],
+  [...]
+}
+```
+
+We added `tpu_version: '2.9.1'` and `accelerator: tpus.v2_8` to the test config and updated the model `command`, removing `--num_gpu` and adding `--distribution_strategy=tpu`. Also note that the input data location (`--data_dir` in this example) [must be in GCS for TPU Nodes](https://cloud.google.com/tpu/docs/storage-options#gcsbuckets). Build the Kubernetes `Job` with `jsonnet`:
+
+```bash
+$ jsonnet --jpath . --string --ext-str gcs-bucket=$GCS_BUCKET doc/tutorials/basic/mnist-tpu.jsonnet
+```
+
+Note the following lines in the generated `Job`:
+
+```
+   [...]
+   "template":
+      "metadata":
+         "annotations":
+            "tf-version.cloud-tpus.google.com": "2.9.1"
+   [...]
+      "resources":
+         "limits":
+            "cloud-tpus.google.com/v2": 8
+```
+
+The TPU software version is specified in `tf-version.cloud-tpus.google.com`. `cloud-tpus.google.com/v2: 8` requests 8 v2 TPU cores, i.e. a v2-8 TPU Node.
+
+If you have sufficient TPU quota, try running the test and observing the output:
+
+```bash
+$ jsonnet --jpath . --string --ext-str gcs-bucket=$GCS_BUCKET doc/tutorials/basic/mnist-tpu.jsonnet | kubectl create -f -
+```
+
+## Next steps
+
+- Build your own Docker container to test your own code. Learn how to [build container images with Cloud Build](https://cloud.google.com/build/docs/building/build-containers).
```

Comment:
I have a question about the best practice - if I want to run unit tests in my repo, which of the two path is more preferrable? 

1. have the docker image only containing the dependencies, and use `git clone` in the command section to get the code, and run tests.
2. Have the docker image contain both the repo and its dependencies, and tests can be directly accessed.